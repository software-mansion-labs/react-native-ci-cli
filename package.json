{
  "name": "react-native-ci-cli",
  "version": "0.3.4",
  "description": "React Native CI CLI is meant to simplify CI/CD setup in your React Native project.",
  "types": "build/types/types.d.ts",
  "bin": {
    "react-native-ci-cli": "bin/react-native-ci-cli"
  },
  "scripts": {
    "clean-build": "rimraf -rf ./build",
    "compile": "tsc -p .",
    "copy-templates": "cp -r ./src/templates/ ./build/templates",
    "build": "yarn clean-build && yarn compile && yarn copy-templates",
    "prepublishOnly": "yarn build",
    "format": "eslint \"**/*.{js,jsx,ts,tsx}\" --fix && prettier \"**/*.{js,jsx,ts,tsx,json}\" --write",
    "test": "jest",
    "watch": "jest --watch",
    "build:alias": "sed -i 's/\"name\": \"react-native-ci-cli\"/\"name\": \"setup-ci\"/g' package.json && yarn build",
    "snapupdate": "jest --updateSnapshot",
    "coverage": "jest --coverage",
    "ts:check": "tsc -p . --noEmit",
    "lint": "eslint \"**/*.{js,jsx,ts,tsx}\"",
    "prettier:check": "prettier --check \"src/**/*.{ts,tsx,js,jsx,json,css,scss,md}\""
  },
  "files": [
    "build",
    "LICENSE",
    "readme.md",
    "bin"
  ],
  "license": "MIT",
  "author": "Maciej Stosio <maciekstosio@users.noreply.github.com>",
  "keywords": [
    "react-native",
    "setup",
    "CI",
    "cli",
    "CD"
  ],
  "dependencies": {
    "@clack/core": "^0.3.4",
    "@clack/prompts": "^0.7.0",
    "gluegun": "5.1.6",
    "hyperlinker": "^1.0.0",
    "is-git-dirty": "^2.0.2",
    "look-it-up": "^2.1.0",
    "pretty-file-tree": "^1.0.1",
    "supports-hyperlinks": "^3.1.0"
  },
  "devDependencies": {
    "@types/jest": "^26.0.20",
    "@typescript-eslint/eslint-plugin": "^4.17.0",
    "@typescript-eslint/parser": "^4.17.0",
    "copyfiles": "^2.4.1",
    "eslint": "^7.22.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-prettier": "^3.3.1",
    "husky": "^5.1.3",
    "jest": "^26.6.3",
    "prettier": "^2.2.1",
    "pretty-quick": "^3.1.0",
    "ts-jest": "^26.5.3",
    "ts-node": "^10.9.1",
    "typescript": "^5.5.3"
  },
  "jest": {
    "preset": "ts-jest",
    "testEnvironment": "node"
  }
}
