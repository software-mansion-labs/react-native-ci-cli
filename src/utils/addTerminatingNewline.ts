import { filesystem } from 'gluegun'

export const addTerminatingNewline = (path: string): void => {
  const content = filesystem.read(path)

  if (!content || content[content.length - 1] !== '\n') {
    filesystem.append(path, '\n')
  }
}
