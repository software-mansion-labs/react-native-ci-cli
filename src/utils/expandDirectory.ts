import { stat, readdir } from 'fs/promises'
import { join } from 'path'

// Recursively finds all files within directory at specified path and returns list of their paths.
export const expandDirectory = async (path: string): Promise<string[]> => {
  const pathStat = await stat(path)

  if (pathStat.isFile()) {
    return [path]
  }

  const files = await readdir(path)

  let result: string[] = []

  for (const file of files) {
    const fullPath = join(path, file)
    const fileStat = await stat(fullPath)

    if (fileStat.isDirectory()) {
      const expanded = await expandDirectory(fullPath)
      result = result.concat(expanded)
    } else {
      result.push(fullPath)
    }
  }

  return result
}
