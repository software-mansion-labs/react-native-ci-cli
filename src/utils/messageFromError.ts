const messageFromError = (e: unknown): string => {
  if (
    typeof e === 'object' &&
    e !== null &&
    'message' in e &&
    typeof e.message === 'string'
  ) {
    return e.message
  }

  // This should normally never be reached, but since you can throw anything
  // and errors have unknown type, we have to handle every case.
  return JSON.stringify(e)
}

export default messageFromError
