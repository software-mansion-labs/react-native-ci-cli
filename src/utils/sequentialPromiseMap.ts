const sequentialPromiseMap = async <T, S>(
  input: T[],
  mapper: (arg: T) => Promise<S>
): Promise<S[]> => {
  const results = []
  for (const next of input) {
    const nextResult = await mapper(next)
    results.push(nextResult)
  }
  return results
}

export default sequentialPromiseMap
