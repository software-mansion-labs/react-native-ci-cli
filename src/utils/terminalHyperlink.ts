const supportsHyperlinks = require('supports-hyperlinks')
const hyperlinker = require('hyperlinker')

export const terminalHyperlink = (text: string, url: string): string => {
  if (supportsHyperlinks.stdout) {
    return hyperlinker(text, url)
  }
  return url
}
