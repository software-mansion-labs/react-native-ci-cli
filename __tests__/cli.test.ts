import { cli } from './utils'
import { version } from '../package.json'

test('prints version', async () => {
  const output = await cli(['--version'])
  expect(output).toContain(version)
})

test('prints help', async () => {
  const output = await cli(['--help'])

  expect(output).toContain(version)

  const outputWithoutVersion = (await output).replace(version, '')
  expect(outputWithoutVersion).toMatchSnapshot()
})
