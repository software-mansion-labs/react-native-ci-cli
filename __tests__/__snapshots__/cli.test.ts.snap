// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prints help 1`] = `
"
[0m[1m[32mWelcome to react-native-ci-cli ![39m[22m[0m

[0m[90mQuickly setup CI workflows for your React Native app[39m[0m

[0m[1m[4mUsage:[24m[22m react-native-ci-cli [FLAGS][0m

[0m[1m[4mFlags:[24m[22m[0m
[0m  [36m--help            [39m	Print help message[0m
[0m  [36m--version         [39m	Print version[0m
[0m  [36m--skip-git-check  [39m	Skip check for dirty git repository[0m
[0m  [36m--preset          [39m	Run with preset. Combine with feature flags to specify generated workflows[0m

[0mUse any combination of the following with --preset flag to specify your own set of workflows to generate[0m

[0m[1m[4mFeature flags:[24m[22m[0m
[0m  [36m--lint            [39m	Generate ESLint workflow to run on every PR[0m
[0m  [36m--jest            [39m	Generate Jest workflow to run on every PR[0m
[0m  [36m--ts              [39m	Generate Typescript check workflow to run on every PR[0m
[0m  [36m--prettier        [39m	Generate Prettier check workflow to run on every PR[0m
[0m  [36m--eas-update      [39m	Generate EAS Update and preview workflow to run on every PR (Expo projects only)[0m
[0m  [36m--detox           [39m	Generate workflow to run Detox e2e tests on every PR[0m

"
`;
